puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/katelloRelease.groovy

```diff
@@ -0,0 +1,156 @@
+pipeline {
+    agent none
+
+    triggers {
+        cron('H 23 * * *')
+    }
+
+    options {
+        timestamps()
+        timeout(time: 2, unit: 'HOURS')
+        disableConcurrentBuilds()
+        ansiColor('xterm')
+    }
+
+	stages {
+		stage('Mash Koji Repositories') {
+	        agent { label 'sshkey' }
+
+			steps {
+
+				sh "ssh -o 'BatchMode yes' root@koji.katello.org katello-mash-split.py"
+
+			}
+		}
+        stage('Client Repoclosure') {
+            steps {
+
+                parallel(
+                    'client/el7': { repoclosure('client', 'el7') },
+                    'client/el6': { repoclosure('client', 'el6') },
+                    'client/fc25': { repoclosure('client', 'f25') },
+                    'client/fc26': { repoclosure('client', 'f26') }
+                )
+
+            }
+        }
+        stage('Pulp Repoclosure') {
+            agent { label 'el' }
+
+            steps {
+
+                repoclosure('pulp', 'el7')
+
+            }
+        }
+        stage('Candlepin Repoclosure') {
+            agent { label 'el' }
+
+            steps {
+
+                repoclosure('candlepin', 'el7')
+
+            }
+        }
+        stage('Katello Repoclosure') {
+            agent { label 'el' }
+
+            steps {
+
+              repoclosure('katello', 'el7')
+
+            }
+        }
+        stage('Install Test') {
+            agent { label 'el' }
+
+            environment {
+                VAGRANT_DEFAULT_PROVIDER = 'rackspace'
+            }
+            steps {
+
+                git url: 'https://github.com/katello/forklift'
+
+                sh "cp -f boxes.yaml.example boxes.yaml"
+                sh "vagrant up centos7-bats-ci"
+
+            }
+            post {
+                always {
+                    sh "mkdir debug"
+                    sh "vagrant ssh-config centos7-bats-ci > ssh_config"
+
+                    sh "scp -F ssh_config centos7-bats-ci:/root/bats_results/*.tap debug/ || true"
+                    sh "scp -F ssh_config centos7-bats-ci:/root/last_logs debug/ || true"
+                    sh "scp -F ssh_config centos7-bats-ci:/root/sosreport* debug/ || true"
+                    sh "scp -F ssh_config centos7-bats-ci:/root/foreman-debug.tar.xz debug/ || true"
+                    sh "scp -F ssh_config centos7-bats-ci:/var/log/foreman-installer/katello.log debug/ || true"
+
+                    sh "vagrant destroy centos7-bats-ci"
+
+                    archive "debug/*"
+                    script { step([$class: "TapPublisher", testResults: "debug/*.tap"]) }
+                    deleteDir()
+                }
+            }
+        }
+        stage('Push RPMs') {
+            agent { label 'admin && sskey' }
+
+            steps {
+
+                sh "ssh -i /var/lib/workspace/workspace/deploy_katello_repos_key/deploy_katello_repos_key katelloproject@fedorapeople.org \"cd /project/katello/bin && sh rsync-repos-from-koji nightly\""
```

Comment:
Can we assume this path exists? The double workspace looks odd as well.

Reply:
Yea this is how the previous job worked. You can see it in other scripts/